src/weather/weather.controller.ts

```diff
@@ -0,0 +1,18 @@
+import { Controller, Get, ParseFloatPipe, Query } from '@nestjs/common';
+import { WeatherService } from './weather.service';
+
+@Controller('weather')
+export class WeatherController {
+  constructor(private readonly weatherService: WeatherService) {}
+
+  @Get()
+  async getForecast(
+    @Query('lat', ParseFloatPipe) lat: number,
+    @Query('lng', ParseFloatPipe) lng: number,
+  ): Promise<WeatherApiResponse | { error: string }> {
+    if (!lat || !lng) {
+      return { error: 'Latitude and longitude are required' };
+    }
+    return await this.weatherService.getForecast(lat, lng);
+  }
```

Comment:
Additional validation logic is needed here I think. Now you're only checking if this value can be parsed to a `number` (specifically to float). You probably want to check whether those are defined with the correct range - lat in (-90, 90) and lng in (-180, 180).

I'd create an additional class for the whole query the user can provide here and use the validation decorators from `class-validator` package. You can also use decorators from `@nestjs/swagger` package so those query params are visible in the swagger documentation as well.

```typescript
export class GetForecastQuery {
  @ApiProperty({ type: 'number', minimum: -90, maximum: 90 })
  @IsNumber()
  @Min(-90)
  @Max(90)
  lat: number;

  @ApiProperty({ type: 'number', minimum: -180, maximum: 180 })
  @IsNumber()
  @Min(-180)
  @Max(180)
  lng: number;
}
```
_(we usually store such classes in `src/<module>/input` directory, so in this case `src/weather/input/get-forecast.query.ts` could be a good place)_

and use this class here like so

```typescript
  @Get()
  async getForecast(
    @Query() query: GetForecastQuery,
  ): Promise<WeatherApiResponse | { error: string }> {
    // we no longer need to think whether `lat` and `lng` are not defined, 
    // class-validator takes care of this and returns BadRequestException if the validation fails
    return this.weatherService.getForecast(query.lat, query.lng);
  }
```